### ISSUE TITLE
ENH: upload - improve coverage 

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandi-cli#214]
Looking at coverage reported by https://github.com/dandi/dandi-cli/pull/210/files seems like many corner cases are missing any testing.  PR could sit on top of #210 which I hope would get merged soonish

**Comment by jwodder** [comment]
Once #217 is merged, [this bit of code](https://github.com/dandi/dandi-cli/blob/bf-win/dandi/upload.py#L120-L127) should be unreachable, because the condition succeeds only if `allow_any_path` is false, in which case the conditional on line 115 will limit `paths` to only containing dandiset metadata files and NWB files.  Should the code be deleted, or should the body of the `if` be replaced with something like:

    raise AssertionError("This shouldn't happen")  # pragma: no cover

**Comment by jwodder** [comment]
Do you just want to cover the parts show in the diff that aren't already covered, or do you want coverage for the whole of the modified files?

For the parts shown in the diff, I'm unsure how to test the following:

* [This block](https://github.com/dandi/dandi-cli/blob/c1daa2322ef2ade67b5ff277d90fab70d6a44fc0/dandi/upload.py#L169-L175) just returns messages to be printed out.  Trying to capture the output would be complicated by the usage of pyout.  `process_path()` cannot be called directly to check its return value, as it is defined inside another function.

* [This block](https://github.com/dandi/dandi-cli/blob/c1daa2322ef2ade67b5ff277d90fab70d6a44fc0/dandi/upload.py#L421-L427) tests for something that isn't supposed to happen.

* [This block](https://github.com/dandi/dandi-cli/blob/c1daa2322ef2ade67b5ff277d90fab70d6a44fc0/dandi/upload.py#L522-L524) just logs & sleeps.

**Comment by yarikoptic** [comment]
in 1st and 3rd -- useful even just to smoke test it so code doesn't blow later on if we change something (expected message keys etc) and it would reach that code path only on a user system.

2nd -- indeed - should not happen... especially with locking now in place.  so it is more of an assert... in this function though I think it might even be worth RFing to avoid exception and just do the same as in the first block -- yield a msg with error and return.  indeed not worth sweating over covering it up in tests, but I would still like to keep that "assertion" in place.

**Comment by yarikoptic** [comment]
> Do you just want to cover the parts show in the diff that aren't already covered, or do you want coverage for the whole of the modified files? 

up to you... sooner or later we would need to improve testing anyways.

> Also, should the PR be merged into bf-win or master?

I have addressed your comments and merged that PR into master to make to progress easier... yet to get to test it in bloody windows "real life" (chasing some other bugs atm)
